### ISSUE TITLE
Add Cracking

### ISSUE THREAD

**Issue by DevonDF** [UndefinedOS/Roadmap#1]
Ideas for cracking:

To make it more realistic, I think we should have two elements.: A Cracker (.crc) and a Scanner (.nmap). With just a Cracker, you can crack into servers that have a firewall with a lower version than yours, but it will take a long time without recon. How to get recon? You use a scanner (.nmap) with cmd 'nmap <host>'. This will return a list of open ports - a better scanner, more ports - but will also list the 'vulnerability state' which is a percentage, of each port. Cracking with an added parameter -p of a correctly vulnerable port will provide the user with a massive time boost.

**Comment by connbob** [comment]
Yeah I like your idea of complexity. Would it be too far if we went ahead and had different types of crackers, or exploits? Maybe a player could research certain exploits somehow to achieve a higher level. 

Potentially, software should be updated from time to time due to randomly generated exploits published on our own exploitDB style page.

Potentially that’s too complex, but it does move away from the basic cracker concept a bit more. Recon would be good nonetheless
